src/components/implementation/no_interface/vkernel/vkernel.c

```diff
@@ -85,6 +85,9 @@ cos_init(void)
 	vk_info.termthd = cos_thd_alloc(vk_cinfo, vk_cinfo->comp_cap, vk_terminate, NULL);
 	assert(vk_info.termthd);
 
+	while (!(cycs = cos_hw_cycles_per_usec(BOOT_CAPTBL_SELF_INITHW_BASE))) ;
```

Comment:
BTW, keep making progress, but you can come back to this review *after* the deadline to fix it up.